devito/petsc/types.py

```diff
@@ -28,21 +28,50 @@ def __init__(self, *args, stencil_width=None, **kwargs):
     def stencil_width(self):
         return self._stencil_width
 
+    @property
+    def info(self):
+        return DMDALocalInfo(name='%s_info' % self.name, liveness='eager')
+
+    @property
+    def _C_free(self):
+        from devito.petsc.utils import petsc_call
+        destroy = petsc_call('DMDestroy', [Byref(self.function)])
+        return destroy
+
 
 class Mat(LocalObject):
     """
     PETSc Matrix object (Mat).
     """
     dtype = CustomDtype('Mat')
 
+    @property
+    def _C_free(self):
+        from devito.petsc.utils import petsc_call
+        destroy = petsc_call('MatDestroy', [Byref(self.function)])
+        return destroy
+
 
-class Vec(LocalObject):
+class LocalVec(LocalObject):
     """
     PETSc Vector object (Vec).
     """
     dtype = CustomDtype('Vec')
 
 
+class GlobalVec(LocalObject):
+    """
+    PETSc Vector object (Vec).
+    """
+    dtype = CustomDtype('Vec')
+
+    @property
+    def _C_free(self):
+        from devito.petsc.utils import petsc_call
+        destroy = petsc_call('VecDestroy', [Byref(self.function)])
```

Comment:
useless assignment

typically we keep the apparently useless assignments if the routine is relatively large , e.g. to easily inspect the variable content with a debugger, or because the assigned variable name encodes information about the variable content, but here it's not very useful